frame/staking/src/lib.rs

```diff
@@ -1484,18 +1520,31 @@ decl_module! {
 		///
 		/// # <weight>
 		/// - The transaction's complexity is proportional to the size of `targets`,
-		/// which is capped at CompactAssignments::LIMIT.
+		/// which is capped at CompactAssignments::LIMIT (MAX_NOMINATIONS).
 		/// - Both the reads and writes follow a similar pattern.
+		/// ---------
+		/// Base Weight: 22.34 + .36 * n µs
+		/// DB Weight:
+		/// - Reads: Era Election Status, Ledger, Current Era
+		/// - Writes: Validators, Nominators
 		/// # </weight>
-		#[weight = 750_000_000]
+		#[weight = FunctionOf(
+			|(targets,): (&Vec<<T::Lookup as StaticLookup>::Source>,)| {
+				T::DbWeight::get().reads_writes(3, 2)
+					.saturating_add(22 * WEIGHT_PER_MICROS)
+					.saturating_add((360 * WEIGHT_PER_NANOS).saturating_mul(targets.len() as Weight))
+			},
+			DispatchClass::Normal,
+			Pays::Yes
+		)]
```

Comment:
This suggestion compiles, not sure what we decided to be prefered
```suggestion
		#[weight = T::DbWeight::get().reads_writes(3, 2)
			.saturating_add(22 * WEIGHT_PER_MICROS)
			.saturating_add((360 * WEIGHT_PER_NANOS).saturating_mul(targets.len() as Weight))
		]
```

Reply:
Where does `targets` come from?

Reply:
as talked offline, not sure my proposed syntax is the way to go even if it works, so anyway fine to keep as is.